### PR TITLE
[Refactor] Update set events functions

### PR DESCRIPTION
**Pull request by vhuseinova-msft**
## Purpose

* Rename set events functions

## Does this introduce a breaking change?

```
[x] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

## What to Check
Verify that the following are valid
* All tests pass
* Events completions are called

### CODE REVIEW

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallComposite.swift [hunk 1]
```diff
@@ -30,15 +30,15 @@ public class CallComposite {
     }
 
     /// Assign closures to execute when error event  occurs inside Call Composite.
-    /// - Parameter didFailAction: The closure returning the error thrown from Call Composite.
-    public func setDidFailHandler(with didFailAction: ((CallCompositeErrorEvent) -> Void)?) {
-        callCompositeEventsHandler.didFail = didFailAction
+    /// - Parameter onErrorHandler: The closure returning the error thrown from Call Composite.
+    public func set(onErrorHandler: ((CallCompositeErrorEvent) -> Void)?) {
+        callCompositeEventsHandler.onError = onErrorHandler
     }
```
[the remark this answers is not in the document]
Comment (mpodwysocki) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallComposite.swift:36]: Or did we want a general `addListener`, `removeListener` regardless of whether they are unicast or multicast?

[the remark this answers is not in the document]
Comment (vhuseinova-msft) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallComposite.swift]: Updated!

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallComposite.swift [hunk 2]
```diff
@@ -10,10 +10,20 @@ import AzureCommunicationCalling
...
+    /// The class to configure events closures for Call Composite.
+    public class Events {
+        /// Closure to execute when error event occurs inside Call Composite.
+        public var onError: ((CallCompositeErrorEvent) -> Void)?
+        /// Closures to execute when participant has joined a call inside Call Composite.
+        public var onRemoteParticipantJoined: (([CommunicationIdentifier]) -> Void)?
+    }
+
+    /// The events handler for Call Composite
+    public let events: Events
```
Comment (tjprescott) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallComposite.swift:23]: Just checking that making this `let` doesn't essentially override the fact that its members are `var`.

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallComposite.swift [hunk 3]
```diff
@@ -53,7 +51,7 @@ public class CallComposite {
 
         dependencyContainer.registerDependencies(callConfiguration,
                                                  localOptions: localOptions,
-                                                 callCompositeEventsHandler: callCompositeEventsHandler)
+                                                 callCompositeEventsHandler: events)
```
Comment (tjprescott) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallComposite.swift:54]: To me `callCompositeEventHandlers` seems like a more accurate name.

**Review by tjprescott — COMMENTED** [review]
Couple comments but overall LGTM.

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallComposite.swift (hunk 2 above)
Comment (vhuseinova-msft) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallComposite.swift:23]: it works well as Events is a class, checked that events closures are called

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallComposite.swift (hunk 3 above)
Comment (vhuseinova-msft) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallComposite.swift]: Updated it to `eventHandler` for now as `callComposite` may be duplication when used in `callComposite. callCompositeEventHandlers`. But will wait if there are any feedback from the team on this as well

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallComposite.swift [hunk 4]
```diff
@@ -12,15 +12,15 @@ import AzureCommunicationCalling
...
+    public class EventsHandler {
         /// Closure to execute when error event occurs inside Call Composite.
         public var onError: ((CallCompositeErrorEvent) -> Void)?
         /// Closures to execute when participant has joined a call inside Call Composite.
         public var onRemoteParticipantJoined: (([CommunicationIdentifier]) -> Void)?
     }
 
     /// The events handler for Call Composite
-    public let events: Events
+    public let eventsHandler: EventsHandler
```
Comment (tjprescott) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallComposite.swift:23]: This was correct before. Should be named `events`.

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Presentation/Manager/RemoteParticipantsManager.swift [hunk 5]
```diff
@@ -12,7 +12,7 @@ protocol RemoteParticipantsManagerProtocol {
 
 class RemoteParticipantsManager: RemoteParticipantsManagerProtocol {
     private let store: Store<AppState>
-    private let eventsHandler: CallComposite.Events
+    private let eventsHandler: CallComposite.EventsHandler
```
Comment (tjprescott) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Presentation/Manager/RemoteParticipantsManager.swift:15]: And the class should be named `Events`

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallComposite.swift [hunk 6]
```diff
@@ -12,15 +12,15 @@ import AzureCommunicationCalling
 public class CallComposite {
 
     /// The class to configure events closures for Call Composite.
-    public class Events {
+    public class EventsHandler {
```
Comment (tjprescott) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallComposite.swift:15]: The previous class name was correct. `Events`.

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallComposite.swift (hunk 4 above)
Comment (vhuseinova-msft) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallComposite.swift:23]: okay, then close [this one](https://github.com/Azure/communication-ui-library-ios/pull/227#discussion_r888210452) as it's about the same property

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallComposite.swift (hunk 3 above)
Comment (vhuseinova-msft) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallComposite.swift]: As discussed [here](https://github.com/Azure/communication-ui-library-ios/pull/227#discussion_r888358963), resolve this conversation

Comment (vhuseinova-msft) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallComposite.swift:54]: Mixed up with the value name. Left it as it is to separate callCompositeEventsHandler and `CallingSDKEventsHandler ` that is created in `registerDependencies` func

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallComposite.swift (hunk 4 above)
Comment (vhuseinova-msft) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallComposite.swift]: Updated

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallComposite.swift (hunk 6 above)
Comment (vhuseinova-msft) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallComposite.swift]: Updated

**Review by jimchou-dev — DISMISSED** [review]

**Review by AmyL219 — DISMISSED** [review]

**Review by kevinyulu — DISMISSED** [review]

**Review by jimchou-dev — APPROVED** [review]

**Review by AmyL219 — APPROVED** [review]

**Pull request merged**